Log message:

General compile fixes for 2.6.16 for sun3, and some updates to make the new bootloader
work correctly.  Tested on 3/50, 3/60, 3/80.

Modified files:
    linux/arch/m68k/kernel:
	sun3-head.S vmlinux-sun3.lds 
    linux/arch/m68k/mm:
	sun3mmu.c 
    linux/include/asm-m68k:
	sun3-head.h sun3ints.h 


Index: linux/arch/m68k/kernel/sun3-head.S
Stats: 12 modifications
http://linux-m68k-cvs.ubb.ca/c/cvsweb/linux/arch/m68k/kernel/sun3%2dhead%2eS.diff?r1=1.6&r2=1.7
========================================================================
--- linux/arch/m68k/kernel/sun3-head.S	5 Apr 2004 13:09:05 -0000	1.6
+++ linux/arch/m68k/kernel/sun3-head.S	11 May 2006 11:14:10 -0000	1.7
@@ -66,17 +66,7 @@
 /* Following code executes at high addresses (0xE000xxx). */
 1:	lea	init_task,%curptr			| get initial thread...
 	lea	init_thread_union+THREAD_SIZE,%sp	| ...and its stack.
-
-/* copy bootinfo records from the loader to _end */
-	lea	_end, %a1
-	lea	BI_START, %a0
-	/* number of longs to copy */
-	movel	%a0@, %d0
-1:	addl	#4, %a0
-	movel   %a0@, %a1@
-	addl	#4, %a1
-	dbf	%d0, 1b
-
+	
 /* Point MSP at an invalid page to trap if it's used. --m */
 	movl	#(PAGESIZE),%d0
 	movc	%d0,%msp
Index: linux/arch/m68k/kernel/vmlinux-sun3.lds
Stats: 2 modifications
http://linux-m68k-cvs.ubb.ca/c/cvsweb/linux/arch/m68k/kernel/vmlinux%2dsun3%2elds.diff?r1=1.1.1.6&r2=1.6
========================================================================
--- linux/arch/m68k/kernel/vmlinux-sun3.lds	9 Apr 2006 16:48:25 -0000	1.1.1.6
+++ linux/arch/m68k/kernel/vmlinux-sun3.lds	11 May 2006 11:14:10 -0000	1.6
@@ -8,7 +8,7 @@
 jiffies = jiffies_64 + 4;
 SECTIONS
 {
-  . = 0xE004000;
+  . = 0xE002000;
   _text = .;			/* Text and read-only data */
   .text : {
 	*(.head)
Index: linux/arch/m68k/mm/sun3mmu.c
Stats: 1 deletion, 6 modifications
http://linux-m68k-cvs.ubb.ca/c/cvsweb/linux/arch/m68k/mm/sun3mmu%2ec.diff?r1=1.9&r2=1.10
========================================================================
--- linux/arch/m68k/mm/sun3mmu.c	28 Jan 2006 21:29:07 -0000	1.9
+++ linux/arch/m68k/mm/sun3mmu.c	11 May 2006 11:14:11 -0000	1.10
@@ -49,7 +49,6 @@
 	unsigned long zones_size[MAX_NR_ZONES] = { 0, };
 	unsigned long size;
 
-
 #ifdef TEST_VERIFY_AREA
 	wp_works_ok = 0;
 #endif
@@ -94,7 +93,11 @@
 	/* memory sizing is a hack stolen from motorola.c..  hope it works for us */
 	zones_size[ZONE_DMA] = ((unsigned long)high_memory - PAGE_OFFSET) >> PAGE_SHIFT;
 
-	free_area_init(zones_size);
+	/* I really wish I knew why the following change made things better...  -- Sam */
+/*	free_area_init(zones_size); */
+	free_area_init_node(0, NODE_DATA(0), zones_size,
+			    (__pa(PAGE_OFFSET) >> PAGE_SHIFT) + 1, NULL);
+
 
 }
 
Index: linux/include/asm-m68k/sun3-head.h
Stats: 1 deletion
http://linux-m68k-cvs.ubb.ca/c/cvsweb/linux/include/asm%2dm68k/sun3%2dhead%2eh.diff?r1=1.2&r2=1.3
========================================================================
--- linux/include/asm-m68k/sun3-head.h	27 Feb 2003 21:20:59 -0000	1.2
+++ linux/include/asm-m68k/sun3-head.h	11 May 2006 11:14:11 -0000	1.3
@@ -4,7 +4,6 @@
 
 #define KERNBASE        0xE000000  /* First address the kernel will eventually be */
 #define LOAD_ADDR       0x4000      /* prom jumps to us here unless this is elf /boot */
-#define BI_START (KERNBASE + 0x3000) /* beginning of the bootinfo records */
 #define FC_CONTROL  3
 #define FC_SUPERD    5
 #define FC_CPU      7
Index: linux/include/asm-m68k/sun3ints.h
Stats: 1 insertion
http://linux-m68k-cvs.ubb.ca/c/cvsweb/linux/include/asm%2dm68k/sun3ints%2eh.diff?r1=1.6&r2=1.7
========================================================================
--- linux/include/asm-m68k/sun3ints.h	28 Jan 2006 21:37:56 -0000	1.6
+++ linux/include/asm-m68k/sun3ints.h	11 May 2006 11:14:11 -0000	1.7
@@ -16,6 +16,7 @@
 #include <asm/intersil.h>
 #include <asm/oplib.h>
 #include <asm/traps.h>
+#include <asm/irq.h>
 
 #define SUN3_INT_VECS 192
 
-
To unsubscribe from this list: send the line "unsubscribe linux-m68k-cvscommit" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

